### PR TITLE
export settings page fixes

### PR DESCRIPTION
**Pull request by DhaaraniCIT**
## Summary by CodeRabbit

- **New Features**
  - Introduced new date export options "Spent Date" and "Current Date" for enhanced reporting flexibility.
  - Enhanced the export settings with the ability to clear selections easily.
  
- **Enhancements**
  - Improved the advanced settings logic for more intuitive user conditions.
  - Updated UI components to reflect new export states and provide clearer information.
  
- **Bug Fixes**
  - Fixed the visibility issue of the clear icon in select fields to show only when applicable.
  
- **Style**
  - Adjusted CSS to improve the visibility and alignment of dropdown trigger icons.
  
- **Documentation**
  - Updated user-facing labels for export settings to better reflect their purposes.

### CODE REVIEW

**Comment by DhaaraniCIT** [comment]
@coderabbitai review

File: src/app/integrations/business-central/business-central-shared/business-central-export-settings/business-central-export-settings.component.ts [hunk 1]
```diff
@@ -90,7 +79,7 @@ export class BusinessCentralExportSettingsComponent implements OnInit {
 
   sessionStartTime = new Date();
 
-  isSaveInProgress: boolean;
+  isSaveInProgress: boolean = false;
```
Comment (ashwin1111) [inline src/app/integrations/business-central/business-central-shared/business-central-export-settings/business-central-export-settings.component.ts:82]: need not initialise this to true by default, this should be true only after cta click

File: src/app/shared/components/configuration/configuration-select-field/configuration-select-field.component.html [hunk 2]
```diff
@@ -54,7 +54,7 @@ <h5 *ngIf="(formControllerName === 'reimbursableExportType' || formControllerNam
                         </div>
                     </ng-template>
                 </p-dropdown>
-                    <div *ngIf="['JOURNAL_ENTRY', 'EXPENSE_REPORT', 'BILL', 'CHARGE_CARD_TRANSACTION', 'PURCHASE_INVOICE', 'DIRECT_COST', 'EXPENSE', 'JOURNAL ENTRY', 'CHECK', 'CREDIT CARD PURCHASE', 'DEBIT CARD EXPENSE'].includes(form.controls[formControllerName].value)" class="sub-text tw-pt-3-px">
+                    <div *ngIf="(appName !== AppName.BUSINESS_CENTRAL || ['JOURNAL_ENTRY', 'PURCHASE_INVOICE'].includes(form.controls[formControllerName].value)) && ['JOURNAL_ENTRY', 'EXPENSE_REPORT', 'BILL', 'CHARGE_CARD_TRANSACTION', 'PURCHASE_INVOICE', 'DIRECT_COST', 'EXPENSE', 'JOURNAL ENTRY', 'CHECK', 'CREDIT CARD PURCHASE', 'DEBIT CARD EXPENSE'].includes(form.controls[formControllerName].value)" class="sub-text tw-text-pretty tw-pt-3-px">
```
Comment (ashwin1111) [inline src/app/shared/components/configuration/configuration-select-field/configuration-select-field.component.html:57]: is this a temp change? also () bracket needs to be handled properly

File: src/app/shared/components/helper/app-landing-page-body/app-landing-page-body.component.ts [hunk 3]
```diff
@@ -20,6 +20,8 @@ export class AppLandingPageBodyComponent implements OnInit {
 
   @Input() embedVideo: string;
 
+  @Input() appName: string;
```
Comment (ashwin1111) [inline src/app/shared/components/helper/app-landing-page-body/app-landing-page-body.component.ts:23]: did we add inputs to all places which uses this component?

File: src/app/shared/components/configuration/configuration-select-field/configuration-select-field.component.html [hunk 4]
```diff
@@ -54,7 +54,7 @@ <h5 *ngIf="(formControllerName === 'reimbursableExportType' || formControllerNam
                         </div>
                     </ng-template>
                 </p-dropdown>
-                    <div *ngIf="(appName !== AppName.BUSINESS_CENTRAL || ['JOURNAL_ENTRY', 'PURCHASE_INVOICE'].includes(form.controls[formControllerName].value)) && ['JOURNAL_ENTRY', 'EXPENSE_REPORT', 'BILL', 'CHARGE_CARD_TRANSACTION', 'PURCHASE_INVOICE', 'DIRECT_COST', 'EXPENSE', 'JOURNAL ENTRY', 'CHECK', 'CREDIT CARD PURCHASE', 'DEBIT CARD EXPENSE'].includes(form.controls[formControllerName].value)" class="sub-text tw-text-pretty tw-pt-3-px">
+                    <div *ngIf="appName !== AppName.BUSINESS_CENTRAL || (['JOURNAL_ENTRY', 'PURCHASE_INVOICE'].includes(form.controls[formControllerName].value)) && ['JOURNAL_ENTRY', 'EXPENSE_REPORT', 'BILL', 'CHARGE_CARD_TRANSACTION', 'PURCHASE_INVOICE', 'DIRECT_COST', 'EXPENSE', 'JOURNAL ENTRY', 'CHECK', 'CREDIT CARD PURCHASE', 'DEBIT CARD EXPENSE'].includes(form.controls[formControllerName].value)" class="sub-text tw-text-pretty tw-pt-3-px">
```
Comment (ashwin1111) [inline src/app/shared/components/configuration/configuration-select-field/configuration-select-field.component.html:57]: stash the change in this file, may be we commented it for demo

**Review by ashwin1111 — APPROVED** [review]

**Pull request merged**